### PR TITLE
add salesforce input plugin

### PR DESCRIPTION
**Pull request by rody**
Gather metrics about limits and usage for a Salesforce organisation.

### Required for all PRs:

- [ X] Signed [CLA](https://influxdata.com/community/cla/).
- [ ] Associated README.md updated.
- [ X] Has appropriate unit tests.

### CODE REVIEW

File: plugins/inputs/salesforce/salesforce.go [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+
+var matchFirstCap = regexp.MustCompile("(.)([A-Z][a-z]+)")
+var matchAllCap = regexp.MustCompile("([a-z0-9])([A-Z])")
+
+// convet a string to snake_case
+func toSnakeCase(str string) string {
+	snake := matchFirstCap.ReplaceAllString(str, "${1}_${2}")
+	snake = matchAllCap.ReplaceAllString(snake, "${1}_${2}")
+	return strings.ToLower(snake)
+}
```
Comment (danielnelson) [inline plugins/inputs/salesforce/salesforce.go]: You can use this function https://github.com/influxdata/telegraf/blob/master/internal/internal.go#L160 for converting to snake case.

File: plugins/inputs/salesforce/salesforce.go [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+		}
+		resp, err = s.queryLimits()
+		if err != nil {
+			return l, err
+		}
+		defer resp.Body.Close()
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return l, fmt.Errorf("Salesforce responded  with unexpected status code %d", resp.StatusCode)
```
Comment (danielnelson) [inline plugins/inputs/salesforce/salesforce.go]: Extra space char in the message

**Comment by rody** [comment]
@danielnelson Thanks for the review!
I've update the pull request accordingly.
I'm not sure if I should squash the 2 commits or keep them separated. Let me know if you want me to squash them.

**Comment by danielnelson** [comment]
Either way is fine, I always merge pull requests with "squash and merge".

**Pull request merged**